packages/core/src/utils/editor.ts

```diff
@@ -114,6 +146,67 @@ export function getDiffCommand(
           newPath,
         ],
       };
+    case 'editor': {
+      // Use VISUAL first, then EDITOR environment variable with appropriate diff handling
+      const editorCmd = process.env.VISUAL || process.env.EDITOR;
+      if (!editorCmd) {
+        return null;
+      }
+      const parts = editorCmd.split(' ');
+      const editorCommand = parts[0];
+      const editorArgs = parts.slice(1);
+      const editorName = editorCommand.toLowerCase();
+
+      // Handle different editors' diff modes
+      if (
```

Comment:
Why special case vi/vim/nvim etc. vs. relying on the user to provide this in the editor args & command?

I ask because this seems to be overloading our existing "pick something specific" and add a little bit too much arcane magic to the editor flow